test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,553 @@
+        # alice spends both of her utxos, replacing bob's tx
+        outputs = {self.charlie.getnewaddress() : alice_v2_unspent['amount'] + alice_unspent['amount'] - Decimal(0.00005120)}
+        alice_txid = self.send_tx(self.alice, [alice_v2_unspent, alice_unspent], outputs, 3)
+        self.sync_mempools()
+        # bob's tx now has a mempool conflict
+        assert_equal(self.bob.gettransaction(bob_txid)['mempoolconflicts'], [alice_txid])
+        # alice fee-bumps her tx so it only spends the v2 utxo
+        outputs = {self.charlie.getnewaddress() : alice_v2_unspent['amount'] - Decimal(0.00015120)}
+        self.send_tx(self.alice, [alice_v2_unspent], outputs, 2)
+        # bob's tx now has non conflicts and can be rebroadcast
+        assert_equal(self.bob.gettransaction(bob_txid)['mempoolconflicts'], [])
+
+    @cleanup
+    def max_tx_weight(self):
+        self.log.info("Test max v3 transaction weight.")
+
+        tx = CTransaction()
+        tx.version = 3 # make this a truc tx
+        # increase tx weight almost to the max truc size
+        self.bulk_tx(tx, 5, TRUC_MAX_VSIZE - 100)
+
+        assert_raises_rpc_error(
+            -4,
+            "The inputs size exceeds the maximum weight. Please try sending a smaller amount or manually consolidating your wallet's UTXOs",
+            self.charlie.fundrawtransaction,
+            tx.serialize_with_witness().hex(),
+            {'include_unsafe' : True}
+        )
+
+        tx.version = 2
+        self.charlie.fundrawtransaction(tx.serialize_with_witness().hex())
+
+    @cleanup
+    def max_tx_child_weight(self):
+        self.log.info("Test max v3 transaction child weight.")
+
+        outputs = {self.alice.getnewaddress() : 10}
+        self.send_tx(self.charlie, [], outputs, 3)
+
+        tx = CTransaction()
+        tx.version = 3
+
+        self.bulk_tx(tx, 5, TRUC_CHILD_MAX_VSIZE - 100)
+
+        assert_raises_rpc_error(
+            -4,
+            "The inputs size exceeds the maximum weight. Please try sending a smaller amount or manually consolidating your wallet's UTXOs",
+            self.alice.fundrawtransaction,
+            tx.serialize_with_witness().hex(),
+            {'include_unsafe' : True}
+        )
+
+        self.generate(self.nodes[0], 1)
+        self.alice.fundrawtransaction(tx.serialize_with_witness().hex())
+
+    @cleanup
+    def user_input_weight_not_overwritten(self):
+        self.log.info("Test that the user-input tx weight is not overwritten by the truc maximum")
+
+        tx = CTransaction()
+        tx.version = 3
+
+        self.bulk_tx(tx, 5, int(TRUC_MAX_VSIZE/2))
+
+        assert_raises_rpc_error(
+            -4,
+            "Maximum transaction weight is less than transaction weight without inputs",
+            self.charlie.fundrawtransaction,
+            tx.serialize_with_witness().hex(),
+            {'include_unsafe' : True, 'max_tx_weight' : int(TRUC_MAX_VSIZE/2)}
```

Comment:
In f4d209c0e94fad8435215abba3e61d79585d0d12 "test: add truc wallet tests"

The user-input tx weight is the same as the previously bulk-ed tx weight?